### PR TITLE
Implement filter GET /assessment

### PR DESCRIPTION
**Pull request by woodwardmw**
This commit implements optional filtering for the GET /assessment endpoint to address issue #433. Users can now filter assessments by revision_id, reference_id, and assessment type using query parameters.

Changes:
- Added optional query parameters (revision_id, reference_id, type) to GET /assessment endpoint
- Updated database queries for both admin and regular users to apply filters when provided
- Maintained backward compatibility - no filters returns all accessible assessments
- Added comprehensive test suite (test_assessment_filtering) covering:
  - Individual filter parameters
  - Multiple filter combinations
  - Admin user filtering
  - Edge cases (no matching results)
  - Backward compatibility verification

The implementation preserves existing authorization logic, ensuring users only see assessments they have permission to access, with optional filtering applied on top of those permission checks.

### CODE REVIEW

**Review by cassiep429 — COMMENTED** [review]
See comment

File: assessment_routes/v3/assessment_routes.py [hunk 1]
```diff
@@ -79,9 +87,17 @@ async def get_assessments(
 
     if current_user.is_admin:
         # Admin users can access all assessments
-        result = await db.execute(
-            select(Assessment).where(Assessment.deleted.is_(False))
-        )
+        stmt = select(Assessment).where(Assessment.deleted.is_(False))
+
```
[the remark this answers is not in the document]
Comment (woodwardmw) [reply assessment_routes/v3/assessment_routes.py]: I think the function signature ensures it's an int. FastAPI will reject the request if it isn't (or can't be coerced into an int).
